pkg/ignore/types.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ignore
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// fileIgnorePattern holds the ignorable patterns
+type fileIgnorePattern struct {
+	paths   []string
+	regExpr string
+	invert  bool
+}
+
+// Ignorable  holds the what type of ignore
+type Ignorable int
```

Comment:
Another trick I have seen with Go "enums" is to define:

```go

// String implements fmt.Stringer
func (i Ignorable) String() string {
   switch i {
      case Current: return "Current"
      case Transitive: return "Transitive"
      case No: return "No"
      default: return "Unknown"
   }
}
```

This lets folks get a friendlier value when they use `%v` to print enum values.

Reply:
this is similar to Java `toString()` overrides 😄